src/ngraph/runtime/reference/topk.hpp

```diff
@@ -63,7 +77,8 @@ namespace ngraph
                       const Shape& out_shape,
                       size_t axis,
                       size_t k,
```

Comment:
As long as you're optimizing this, you might as well move the coordinate iterator loop inside the tests for sort, etc. instead of doing all those tests each time.